### PR TITLE
HtmlAttachments sync check

### PR DESCRIPTION
**Pull request by gpeng**
This adds support for the `HtmlAttachment` -> `HtmlPublication` format to the sync checks as part of its migration.

[Trello](https://trello.com/c/uwsLR1xU/285-html-publications-migration-implement-publishing-of-format-to-publishing-api-large)

### CODE REVIEW

File: lib/sync_checker/checks/html_attachment_unpublished_check.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+      def there_is_an_item_in_the_content_store?
+        response.body != ""
+      end
+
+      def attachable_has_been_withdrawn?
+        attachable.withdrawn?
+      end
+
+      def check_for_withdrawn_notice#(unpublishing, content_item)
```
Comment (andrewgarner) [inline lib/sync_checker/checks/html_attachment_unpublished_check.rb]: We shouldn't commit commented out code.

File: lib/sync_checker/checks/html_attachment_unpublished_check.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      def check_redirected
+        if attachable_should_be_redirected_to_parent?
+          check_for_redirect_to_parent
+        else
+          check_for_redirect_to_alternative_url
+        end
+      end
+
+      def attachable_should_be_redirected_to_parent?
+        unpublishing.unpublishing_reason_id == 1 &&
```
Comment (andrewgarner) [inline lib/sync_checker/checks/html_attachment_unpublished_check.rb]: `1` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        )
+
+        assert_equal [], HtmlAttachmentUnpublishedCheck.new(attachment).call(stub)
+      end
+
+      def test_returns_no_errors_if_the_attachable_is_withdrawn_and_the_attachment_has_a_withdrawn_notice
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 5,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `5` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+        @stub_renderer.stubs(:govspeak_to_html).returns("<p>Withdrawnificated</p>")
+
+        assert_equal [], HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_an_error_if_the_document_should_by_withdrawn_but_has_no_notice
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 5,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `5` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+        assert_equal [expected_error],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_no_error_if_the_attachable_is_unpublished_in_error_and_the_attachment_returns_a_redirect_to_the_parent
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 1,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `1` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+        assert_equal [],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_an_error_if_the_attachable_is_unpublished_in_error_and_there_is_no_redirect_to_the_parent
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 1,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `1` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+
+        assert_equal [expected_error],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_no_error_if_the_attachable_is_unpublished_in_error_and_the_attachment_returns_a_redirect_to_the_alternative_url
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 1,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `1` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+
+        assert_equal [],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_an_error_if_the_attachable_is_unpublished_in_error_and_the_attachment_does_not_redirect
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 1,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `1` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+
+        assert_equal [expected_error],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_no_error_if_attachable_consolidated_and_attachment_redirects
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 4,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `4` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+
+        assert_equal [],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_an_error_if_attachable_consolidated_and_attachment_does_not_redirect
+        attachment = stub(
+          attachable: stub(
+            unpublishing: stub(
+              unpublishing_reason_id: 4,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `4` is a magic number.

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+        assert_equal [expected_error],
+          HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+    end
+
+    def test_returns_an_error_if_there_is_no_item_in_the_content_store
+      attachment = stub(
+        attachable: stub(
+          unpublishing: stub(
+            unpublishing_reason_id: 4,
```
Comment (andrewgarner) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: `4` is a magic number.

File: lib/sync_checker/checks/html_attachment_unpublished_check.rb (hunk 1 above)
Comment (gpeng) [inline lib/sync_checker/checks/html_attachment_unpublished_check.rb]: oops

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+            }
+          }.to_json
+        )
+
+        @stub_renderer.stubs(:govspeak_to_html).returns("<p>Withdrawnificated</p>")
+
+        assert_equal [], HtmlAttachmentUnpublishedCheck.new(attachment).call(response)
+      end
+
+      def test_returns_an_error_if_the_document_should_by_withdrawn_but_has_no_notice
```
Comment (bevanloon) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: is `by` a typo? should it be `be`?

File: lib/sync_checker/checks/html_attachment_unpublished_check.rb (hunk 1 above)
Comment (gpeng) [inline lib/sync_checker/checks/html_attachment_unpublished_check.rb]: Fixed in ae4021c

File: lib/sync_checker/checks/html_attachment_unpublished_check.rb (hunk 2 above)
Comment (gpeng) [inline lib/sync_checker/checks/html_attachment_unpublished_check.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 3 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 4 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 5 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 6 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 7 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 8 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 9 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

File: test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb (hunk 12 above)
Comment (gpeng) [inline test/unit/sync_checker/checks/html_attachment_unpublished_check_test.rb]: Fixed in ae4021c

**Pull request merged**